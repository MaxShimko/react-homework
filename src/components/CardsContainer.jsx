import React from 'react';
import { Container, Row, Col } from 'react-bootstrap';

import '../styles/CardsContainer.scss';
import CardItem from './CardItem.jsx';

class CardsContainer extends React.Component {
    constructor(props) {
        super(props);

        this.allMovies = [];
        this.releaseTimestampSymbol = Symbol('release timestamp');
        this.combinedGenreSymbol = Symbol('combined genre');

        this.state = {
            allMovies: []
        };
    }

    async componentDidMount() {
        try {
            const allMovies = (await fetch('https://reactjs-cdp.herokuapp.com/movies')
                .then(res => res.json())).data;

            allMovies.forEach(movie => {
               movie[this.releaseTimestampSymbol] = +new Date(movie.release_date);
               movie[this.combinedGenreSymbol] = movie.genres.join(' ');
            });

            // console.log(allMovies);

            this.setState({
                allMovies
            });
        } catch (e) {
            console.error(e);
        }
    }

    getFilteredMovies() {
        const filterField = ['title', this.combinedGenreSymbol][this.props.filter.type];
        const sortField = [this.releaseTimestampSymbol, 'vote_count'][this.props.sortType];

        return this.state.allMovies
            .filter(movie => movie[filterField].toLowerCase().includes(this.props.filter.string.toLowerCase()))
            .sort((a, b) => b[sortField] - a[sortField]);
    }

    render() {
        const movies = this.getFilteredMovies();

        return <div className="main-page-content">
            <Container>
                <Row className="cards-container">
                    {movies.map((value, index) => {
                        return <Col key={value.id} xs="12" md="6" lg="4" className="card-item-wrapper">
                            <CardItem
                                posterUrl={value.poster_path}
                                title={value.title}
                                ganre={value.genres.join(' & ')}
                                year={value.release_date.split('-')[0]}
                            />
                        </Col>;
                    })}
                    {!movies.length ?
                        (<div className="empty-movies">
                            <div>No Films Found</div>
                        </div>)
                        : null}
                </Row>
            </Container>
        </div>;
    }
}

export default CardsContainer;
