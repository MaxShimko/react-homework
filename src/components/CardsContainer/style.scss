.main-page-content {
  .cards-container {
    padding: 20px;
    background: #232323;

    .card-item-wrapper {
      padding: 0;
    }

    .empty-movies {
      width: 100%;
      height: 400px;
      display: flex;
      justify-content: center;
      align-items: center;

      div {
        font-size: 40px;
        color: rgba(#fff, 0.8);
      }
    }
  }
}
