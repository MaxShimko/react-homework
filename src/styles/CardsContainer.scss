.main-page-content {
  .cards-container {
    padding: 20px;
    background: #232323;

    .card-item-wrapper {
      padding: 0;
    }
  }
}
