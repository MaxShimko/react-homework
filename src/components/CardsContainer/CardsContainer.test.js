import React from 'react';
import { shallow } from 'enzyme';
import CardsContainer from './CardsContainer';
import {describe, expect, it} from '@jest/globals';
import toJson from 'enzyme-to-json';

describe('<CardsContainer />', () => {
    it('renders without crashing', () => {
        const component = shallow(<CardsContainer filter={{type: 0, string: ''}} sortType={0} />);
        expect(toJson(component)).toMatchSnapshot();
    });
});
