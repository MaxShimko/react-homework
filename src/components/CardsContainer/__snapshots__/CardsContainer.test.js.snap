// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardsContainer /> renders without crashing 1`] = `
<div
  className="main-page-content"
>
  <Container
    fluid={false}
  >
    <Row
      className="cards-container"
      noGutters={false}
    >
      <div
        className="empty-movies"
      >
        <div>
          No Films Found
        </div>
      </div>
    </Row>
  </Container>
</div>
`;
